bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigTomlProvider.java

```diff
@@ -248,6 +251,13 @@ private TomlNode getTomlNode(TomlNode tomlValue, String variableName, Type type)
     }
 
     private static TomlTableNode getConfigurationData(Path configFilePath, boolean hasRequired) {
+        TomlTableNode tomlNode;
+        if (configFilePath.equals(DEFAULT_CONFIG_PATH)) {
```

Comment:
Accrding to the spec if BAL_CONFIG_FILES not specified we need to check config_data. but with this approach even I have specified BAL_CONFIG_FILES we check the CONFIG_DATA.

IMO this logic needs to go to LaunchUtils. We need to have ConfigTomlProvider constructor which takes string data and getConfigurationData or TomlFileProvider && TomlContentProvider. This logic should be outside